clang/test/CIR/CodeGen/basic.cpp

```diff
@@ -102,3 +102,20 @@ size_type max_size() {
 // CHECK:   %3 = cir.cast(integral, %2 : !s32i), !u64i
 // CHECK:   %4 = cir.const #cir.int<8> : !u64i
 // CHECK:   %5 = cir.binop(div, %3, %4) : !u64i
+
+void ref_arg(int &x) {
+  int y = x;
+  x = 3;
+}
```

Comment:
Can we also have tests for 'function that returns reference'?
```
ClassTy &foo();
ClassTy &&foo();
```

And perhaps `int &z = x`?

Reply:
Sure

Reply:
After looking into it a bit more, I don't want to handle r-value references yet. It needs more handling. The other cases are added along with the little extra support they required.

Reply:
OK, sounds good. Please change the  title then to say "Upstream support for L-Value References"